spec/models/repository/git_spec.rb

```diff
@@ -0,0 +1,362 @@
+          changesets = instance.latest_changesets('', nil, 2)
+          expect(changesets.size).to eq(2)
+        end
+
+        it 'should fetch changesets with paths' do
+          changesets = instance.latest_changesets('images', nil)
+          expect(changesets.map(&:revision))
+            .to eq([
+                  'deff712f05a90d96edbd70facc47d944be5897e3',
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+
+          changesets = instance.latest_changesets('README', nil)
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '32ae898b720c2f7eec2723d5bdd558b4cb2d3ddf',
+                  '4a07fe31bffcf2888791f3e6cbc9c4545cefe3e8',
+                  '713f4944648826f558cf548222f813dabe7cbb04',
+                  '61b685fbe55ab05b5ac68402d5720c1a6ac973d1',
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+        end
+        
+        it 'should fetch changesets with path, revision and limit' do
+          changesets = instance.latest_changesets('images', '899a15dba')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                ])
+
+          changesets = instance.latest_changesets('images', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', '899a15dba')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+        end
+
+        it 'should fetch changesets with tag' do
+          changesets = instance.latest_changesets('images', 'tag01.annotated')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
+
+          changesets = instance.latest_changesets('README', '899a15dba', 1)
+          expect(changesets.map(&:revision))
+            .to eq(['899a15dba03a3b350b89c3f537e4bbe02a03cdc9'])
+
+          changesets = instance.latest_changesets('README', 'tag01.annotated')
+          expect(changesets.map(&:revision))
+            .to eq([
+                  '899a15dba03a3b350b89c3f537e4bbe02a03cdc9',
+                  '7234cb2750b63f47bff735edc50a1c0a433c2518'
+                  ])
```

Comment:
Indent the right bracket the same as the start of the line where the left bracket is.